test/Microsoft.NET.Build.Tests/GivenThatWeWantToTargetNet471.cs

```diff
@@ -127,14 +159,120 @@ public void It_does_not_include_facades_from_nuget_packages()
             outputDirectory.Should().OnlyHaveFiles(new[] {
                 $"{testProject.Name}.exe",
                 $"{testProject.Name}.pdb",
-
-                //  Remove these two once https://github.com/dotnet/sdk/issues/1647 is fixed
+                
                 "System.Net.Http.dll",
                 "System.IO.Compression.dll",
 
                 //  This is an implementation dependency of the System.Net.Http package, which won't get conflict resolved out
                 "System.Diagnostics.DiagnosticSource.dll",
             });
         }
+
+        [Fact]
+        public void It_includes_shims_when_net471_app_references_netstandard16()
+        {
+            //  https://github.com/dotnet/sdk/issues/1625
+            if (!Net471ReferenceAssembliesAreInstalled())
+            {
+                return;
+            }
+            var testProject = new TestProject()
+            {
+                Name = "Net471App_Referencing_NetStandard16",
+                TargetFrameworks = "net471",
+                IsSdkProject = true,
+                IsExe = true
+            };
+
+            var netStandardProject = new TestProject()
+            {
+                Name = "NetStandard16_Library",
+                TargetFrameworks = "netstandard1.6",
+                IsSdkProject = true
+            };
+
+            testProject.ReferencedProjects.Add(netStandardProject);
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, "net471_ref_ns16")
+                .Restore(Log, testProject.Name);
+
+            var buildCommand = new BuildCommand(Log, Path.Combine(testAsset.TestRoot, testProject.Name));
+
+            buildCommand
+                .Execute("/v:normal")
+                .Should()
+                .Pass()
+                .And.NotHaveStdOutContaining("MSB3277") // MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.
+                .And.NotHaveStdOutContaining("Could not determine");
+
+            var outputDirectory = buildCommand.GetOutputDirectory(testProject.TargetFrameworks);
+
+            outputDirectory.Should().OnlyHaveFiles(new[] {
+                $"{testProject.Name}.exe",
+                $"{testProject.Name}.pdb",
+                $"{netStandardProject.Name}.dll",
+                $"{netStandardProject.Name}.pdb",
+                "System.Diagnostics.DiagnosticSource.dll" // This library will get pulled in as part of the closure of the ns16 project and will be copyied because it's not inbox.
```

Comment:
@AlexGhiondea I believe this might actually be a bug on the framework, so we might want to include this file inbox on 472.

Reply:
Don't blindly include that inbox. Please consult with @vancem first as I know they were trying to keep it out of box.

Reply:
I'd use the same comment here as in the previous test: `//  This is an implementation dependency of the System.Net.Http package, which won't get conflict resolved out`

I'm also pretty sure we don't want to include this inbox.  It's an implementation dependency of the OOB version of System.Net.Http.  Conflict resolution will pick the support package of the Http library, but it doesn't know that this dependency is no longer needed.

If you want to consider making it so that this library isn't copied, please file a bug for that in the sdk repo.

Reply:
Sounds good I'll update it as requested.